### PR TITLE
add ssh key so that an API created host can be accessed by ssh

### PR DESCRIPTION
**Pull request by xunpan**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by xunpan** [comment]
By the way, the API created host can access internet by default based on my test. 
So, I think the created machine is ready for run Kubernetes deployment script to deployment kubernetes.

File: pkg/cloud/ibmcloud/clients/machines.go [hunk 1]
```diff
@@ -104,3 +114,25 @@ func (gs *GuestService) GuestList() ([]datatypes.Virtual_Guest, error) {
...
+		log.Printf("Error retrieving ssh keys from Account: %s\n", err)
+		return id
+	}
+
+	for _, key := range keys {
+		if *key.Label == name {
+			id = *key.Id
+			log.Printf("Get SSH key for %q with value %d\n", *key.Label, *key.Id)
+			break
+		}
```
Comment (jichenjc) [inline pkg/cloud/ibmcloud/clients/machines.go]: do we want to add a error case here to know the key is not there? or you will give some error like 'id 0' not found which will be weird?

**Comment by xunpan** [comment]
Thanks @jichenjc for you comments. Addressed.

**Comment by jichenjc** [comment]
code is good, let's wait for CI issue fix then merge this

**Comment by gyliu513** [comment]
CI failed @xunpan @morvencao https://travis-ci.com/multicloudlab/cluster-api-provider-ibmcloud/builds/108927458

```
+echo 'Downloading test dependencies'
Downloading test dependencies
+download-dependencies
+go get -u golang.org/x/lint/golint
+[[ -z y ]]
+./scripts/download-binaries.sh
./scripts/pre-commit.sh: line 39: ./scripts/download-binaries.sh: Permission denied
The command "test $DOC_ONLY && echo "Skipped!" || DOWNLOAD_BINARIES=y bash -x ./scripts/pre-commit.sh" exited with 1.
```

**Comment by morvencao** [comment]
@xunpan Would you add the `execute` permission for the scripts under `hack` and `scripts`?

**Comment by xunpan** [comment]
OK. I will guarantee the CI succeed in this ticket.

**Pull request merged**